core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPageSourceProvider.java

```diff
@@ -28,4 +30,31 @@ ConnectorPageSource createPageSource(
             ConnectorTableHandle table,
             List<ColumnHandle> columns,
             DynamicFilter dynamicFilter);
+
+    /**
+     * Returns unenforced (effective) predicate that {@link ConnectorPageSource} would use to filter split data.
+     * If split is completely filtered out, then this method should return {@link TupleDomain#none}.
+     */
+    default TupleDomain<ColumnHandle> getUnenforcedPredicate(
+            ConnectorSession session,
+            ConnectorSplit split,
+            ConnectorTableHandle table,
+            TupleDomain<ColumnHandle> dynamicFilter)
```

Comment:
I though about it, but I think it's less confusing when the arg is actually named `dynamic filter`. We could pass `DynamicFilter` itself, but I think it's overkill.

Some connectors won't simplify dynamic filters because they will use them for indexing.